src/consumers.ts

```diff
@@ -187,8 +244,38 @@ Recommendation: ${recommendedVersion}`;
                 source: '\nDependency Analytics Plugin [Powered by Snyk]',
             };
 
+            let diagnosticLinePresent = false
+            let diagnosticIndex = 0
+            for (let i = 0; i < cls.diagnostics.length; i++) {
+                if (cls.diagnostics[i].range.start.line.toString() == diagnostic.range.start.line.toString()) {
+                    diagnosticLinePresent = true
+                    diagnosticIndex = i
+                }
+            }
+
+            if (diagnosticLinePresent) {
+                // Add new counts to old one
+                const oldMessage = cls.diagnostics[diagnosticIndex].message.split("\n")
+                const packageCount = parseInt(oldMessage[1].split(":")[1].trim()) + 1
+                const securityVulnerabilityCount = parseInt(oldMessage[2].split(":")[1].trim()) + this.vulnerabilityCount
+                const securityAdvisoryCount = parseInt(oldMessage[3].split(":")[1].trim()) + this.advisoryCount
+                const newExploitCount = this.getAggregaterExploitCount(oldMessage[4].split(":")[1].trim())
+                const newSeverity = this.getAggregatedSeverity(oldMessage[5].split(":")[1].trim())
+                const newRecVersion = this.getMaxRecVersion(oldMessage[6].split(":")[1].trim())
```

Comment:
Text based parsing to derive the aggregated response will not be scalable. Please maintain a global state/store which can act as an aggregator for Diagnostic messages.

Reply:
Added another package aggregator store to merge module and packages together.